python/ray/train/v2/api/data_parallel_trainer.py

```diff
@@ -225,6 +222,33 @@ def fit(self) -> Result:
 
         return result
 
+    def _initialize_controller(
+        self, **controller_init_kwargs
+    ) -> Union[TrainController, ActorHandle]:
+        if self._run_controller_as_actor:
+            # Attach the controller to the node running the driver script.
+            controller_actor_cls = ray.remote(
+                num_cpus=0,
+                scheduling_strategy=NodeAffinitySchedulingStrategy(
+                    node_id=ray.get_runtime_context().get_node_id(), soft=False
+                ),
+                runtime_env={"env_vars": get_env_vars_to_propagate()},
```

Comment:
Unrelated to this PR, but would this merge with environment variables set by a user-defined runtime env?

Reply:
I don't know actually, I'll test it out